File: cpp/problems/syntax/syntax026.cpp
#include <iostream>
using namespace std;

int main(int argc, char *argv[]) {
    int b = 1000;
    cout << (b >> 8) << endl; // 3 * 256 = 768 ; 4 * 256 = 1024
    cout << (768 >> 8) << endl; // 3
    cout << (767 >> 8) << endl;  // 2
    cout << (1023 >> 8) << endl; // 3
    cout << (1024 >> 8) <<endl; // 4
    return 0;
}

File: objective-c/syntax/syntax026.m
#include <Foundation/Foundation.h>

int main(int argc, char *argv[]) {
    int (^power)(int, int) = ^(int base, int pew) {
        int result = 1;
        while(pew--) result *= base;
        return result;
    };
    int a = power(3, 3);
    printf("%i\n", a);
    a = power(2, 3);
    printf("%i\n", a);
    a = power(5, 3);
    printf("%i\n", a);
    return 0;
}
